pkg/manifests/manifests.go

```diff
@@ -2119,6 +2160,52 @@ func (f *Factory) MetricsServerDeployment(kubeletCABundle *v1.ConfigMap, tlsSecr
 	return dep, nil
 }
 
+func (f *Factory) MetricsServerSecret(tlsSecret *v1.Secret, apiAuthConfigmap *v1.ConfigMap) (*v1.Secret, error) {
+	data := make(map[string]string)
+
+	for k, v := range tlsSecret.Data {
+		data[k] = string(v)
+	}
+
+	for k, v := range apiAuthConfigmap.Data {
+		data[k] = v
+	}
+
+	r := newErrMapReader(data)
+
+	var (
+		clientCA              = r.value("client-ca-file")
+		requestheaderClientCA = r.value("requestheader-client-ca-file")
+		tlsCA                 = r.value("tls.crt")
+		tlsKey                = r.value("tls.key")
+	)
+
+	if r.Error() != nil {
+		return nil, fmt.Errorf("value not found in extension api server authentication configmap: %w", r.err)
+	}
+
+	h := fnv.New64()
+	h.Write([]byte(clientCA + requestheaderClientCA + tlsCA + tlsKey))
+	hash := strconv.FormatUint(h.Sum64(), 32)
+
+	return &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: f.namespace,
+			Name:      fmt.Sprintf("metrics-server-%s", hash),
+			Labels: map[string]string{
+				"monitoring.openshift.io/name": "metrics-server",
+				"monitoring.openshift.io/hash": hash,
+			},
+		},
+		Data: map[string][]byte{
+			"client-ca-file":               []byte(clientCA),
+			"requestheader-client-ca-file": []byte(requestheaderClientCA),
+			"tls.crt":                      []byte(tlsCA),
+			"tls.key":                      []byte(tlsKey),
```

Comment:
I don't see where we mount/set these two.

Reply:
https://github.com/openshift/cluster-monitoring-operator/blob/900ffda2f64b256c739679a90555e896033dca7c/assets/metrics-server/deployment.yaml#L85-L86